### ISSUE TITLE
Add optional Gated Attention mechanism from NeurIPS 2025 Best Paper to MultiHeadAttention layer

### ISSUE THREAD

**Issue by pass-lin** [keras-team/keras#22337]
**Feature Request**  
I would like to propose adding an optional gating mechanism to `keras.layers.MultiHeadAttention`, based on the recent **NeurIPS 2025 Best Paper** titled *“Gated Attention for Large Language Models: Non-linearity, Sparsity, and Attention-Sink-Free”* ([link](https://blog.neurips.cc/2025/11/26/announcing-the-neurips-2025-best-paper-awards/)).  

The paper shows that applying a **head‑specific sigmoid gate** after the Scaled Dot‑Product Attention (SDPA) output consistently improves performance, training stability, and eliminates “attention sink”. The modification is minimal yet effective – it is already used in models like Qwen3‑Next.  

**Proposed change**  
Add a boolean parameter (e.g., `use_gate=False`) to `MultiHeadAttention`. When enabled, after computing the SDPA output for each head, we apply:  

```python
gate = ops.sigmoid(keras.layers.Dense(units=head_dim, use_bias=False)(query))  # shape (batch, seq_len, head_dim)
output = output * gate
```  

If this feature is approved, I’m happy to submit a PR in the near future. Thanks for considering!

**Comment by jeffcarp** [comment]
Thanks for the feature request, this sounds good to add
